controllers/pubmed.controller.ts

```diff
@@ -0,0 +1,212 @@
+                    }
+                }
+            }
+        })
+        .catch((error) => {
+            return {
+                statusCode: error.status,
+                statusText: error
+            }
+        });
+}
+function retrieveFirstNew100PubMedArticles(pubMedData: any , featureGeneratorData: any)
+{
+      console.log('coming into retrieveFirstNew100Publications1************************',pubMedData);
+      console.log('coming into retrieveFirstNew100Publications2************************',featureGeneratorData);
+        
+      let finalPubMedArtickes={}; 
+      let filterPubMedArticles = []; 
+      let filter100PubMedArticles = [];
+      let accpetedPubMedCount =0;
+      let rejectedPubMedCount =0;
+     
+    if(pubMedData && featureGeneratorData && featureGeneratorData.statusText && featureGeneratorData.statusText.reciterData && featureGeneratorData.statusText.reciterData.reCiterArticleFeatures 
+        && featureGeneratorData.statusText.reciterData.reCiterArticleFeatures.length > 0)
+    {   
+        console.log('inside if*********************************');
+         for(let pubMedPMIDIndex=0; pubMedPMIDIndex < pubMedData.length ; pubMedPMIDIndex++) 
+        //for(let fgPmidIndex =0; fgPmidIndex < featureGeneratorData.statusText.reciterData.reCiterArticleFeatures.length; fgPmidIndex++ )     
+        {
+            let pmidFound =false;
+            let pubMedPmid = pubMedData[pubMedPMIDIndex].medlinecitation.medlinecitationpmid.pmid;
+           // let  fgPmid = featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].pmid ;
+            
+            for(let fgPmidIndex =0; fgPmidIndex < featureGeneratorData.statusText.reciterData.reCiterArticleFeatures.length; fgPmidIndex++ )    
+             // for(let pubMedPMIDIndex=0; pubMedPMIDIndex < pubMedData.length ; pubMedPMIDIndex++)  
+              {
+                if(featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].pmid== pubMedPmid
+                    &&(featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].userAssertion ==='ACCEPTED' ||
+                    featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].userAssertion ==='REJECTED')
+                    )
+                {
+                    pmidFound = true;
+                    if(featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].userAssertion ==='ACCEPTED')
+                    {
+                        accpetedPubMedCount++;
+                    }
+                    else if(featureGeneratorData.statusText.reciterData.reCiterArticleFeatures[fgPmidIndex].userAssertion ==='REJECTED')
+                    {
+                        rejectedPubMedCount++;
+                    }
+                    break;
+                }
+            }
+            if(!pmidFound)
+            {
+                console.log('inside else block*******************************');
```

Comment:
careful again not to push debug logic. Also let me know in the future. We can setup a debugger to make this much easier

Reply:
removed.